HungarianAlgorithm/Hungarian/Solution.cs

```diff
@@ -0,0 +1,28 @@
+﻿namespace Hungarian
+{
+    public class Solution
+    {
+        public readonly IEnumerable<WellAssignments> Assignments;
+
+        public Solution(IEnumerable<WellAssignments> assignments)
+        {
+            Assignments = assignments;
+        }
+
+        public void Write(TextWriter? textWriter = null)
```

Comment:
Zastanawiam się czy tej metody całkiem nie przenieść do klasy `FileWriter`? Bo teraz wypisywanie jest w 2 miejscach

Reply:
Właśnie się zastanawiałem nad tym. Tylko chciałem, żeby `TextWriter` było przyjmowane jako input, żeby móc wypisywać w razie potrzeby rozwiązanie również na konsolę. A jak przeniesiemy tę metodę do `FileWritera` to wtedy trochę nienaturalnie będziemy korzystać z klasy do zapisu do pliku w tym kontekście. Ale możemy o tym pogadać jeszcze :)

Reply:
Oki, rozumiem :) Ale z tego co widzę, to `StreamWriter` dziedziczy po `TextWriter` więc coś takiego teraz też działa:
```cs
using var writer = new StreamWriter($"{pathToOutputDirectory}/{filename}");
solution.Write(writer);
```
wtedy w sumie można nawet bez `FileWriter`a się obyć

Reply:
@daveedoo Tak, właściwie to `FileWriter` służy nam wyłącznie jako wrapper na te dwie linie i wyizolowanie zapisu do pliku w jedno osobne miejsce niż właściwy algorytm czy sam program:
```
using var writer = new StreamWriter($"{pathToOutputDirectory}/{filename}");
solution.Write(writer);
```
Wtedy gdyby tutaj coś poszło nie tak i chcielibyśmy obsłużyć jakiś błąd albo cokolwiek zmienić to modyfikujemy sam `FileWriter`

Reply:
@Michalski-Artur no dobra, chyba jest git 😉